onnxruntime/test/python/quantization/test_save_activations.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+# coding: utf-8
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""Testing save_activations module
+"""
+
+import tempfile
+import unittest
+from pathlib import Path
+
+import numpy as np
+import onnx
+from onnx import TensorProto, helper, numpy_helper
+
+import onnxruntime
+from onnxruntime.quantization.calibrate import CalibrationDataReader
+from onnxruntime.quantization.save_activations import aug_model_for_act_saving, run_collect_activations
+
+
+def generate_input_initializer(tensor_shape, tensor_dtype, input_name):
+    """
+    Helper function to generate initializers for test inputs
+    """
+    tensor = np.random.normal(0, 0.3, tensor_shape).astype(tensor_dtype)
+    init = numpy_helper.from_array(tensor, input_name)
+    return init
+
+
+class TestDataReader(CalibrationDataReader):
+    """Random Data Input Generator
+    """
+
+    def __init__(self):
+        self.preprocess_flag = True
+        self.enum_data_dicts = []
+        self.count = 2
+        self.input_data_list = []
+        for _ in range(self.count):
+            self.input_data_list.append(np.random.normal(0, 0.33, [1, 3, 1, 3]).astype(np.float32))
+
+    def get_next(self):
+        if self.preprocess_flag:
+            self.preprocess_flag = False
+            input_name = "input"
+            self.enum_data_dicts = iter([{input_name: input_data} for input_data in self.input_data_list])
+        return next(self.enum_data_dicts, None)
+
+    def rewind(self):
+        self.preprocess_flag = True
+
+
+class TestSaveActivations(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls._tmp_model_dir = tempfile.TemporaryDirectory(prefix="test_save_activations.")
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._tmp_model_dir.cleanup()
+
+    def construct_test_model1(self, test_model_path):
```

Comment:
Looks like the function does not require access to the class state. Can it be a function in the module namespace (global in test_save_activations.py)?

Reply:
fixed